gen/maxsize.go

```diff
@@ -0,0 +1,375 @@
+package gen
+
+import (
+	"bytes"
+	"fmt"
+	"go/ast"
+	"io"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/algorand/msgp/msgp"
+)
+
+type maxSizeState uint8
+
+const (
+	// need to write "s = ..."
+	assignM maxSizeState = iota
+
+	// need to write "s += ..."
+	addM
+
+	// can just append "+ ..."
+	exprM
+
+	multM
+	// the result is multiplied by whatever is preceeding it
+)
+
+func maxSizes(w io.Writer, topics *Topics) *maxSizeGen {
+	return &maxSizeGen{
+		p:      printer{w: w},
+		state:  assignM,
+		topics: topics,
+	}
+}
+
+type maxSizeGen struct {
+	passes
+	p      printer
+	state  maxSizeState
+	ctx    *Context
+	topics *Topics
+}
+
+func (s *maxSizeGen) Method() Method { return MaxSize }
+
+func (s *maxSizeGen) Apply(dirs []string) error {
+	return nil
+}
```

Comment:
I assume this function exists to satisfy an Interface? If so could you reference said interface in a comment `// Apply implements ...`